File: src/main/scala/otserver4s/utils/SecurityUtils.scala
package otserver4s.utils

object MD5Utils {
  private val messageDigest = java.security.MessageDigest.getInstance("MD5")
  private val SALT = otserver4s.PropriedadeConfiguracoes("senha.salt")
  def str2md5(senha: String) = {
    messageDigest.digest(senha.concat(SALT)
      .getBytes(java.nio.charset.StandardCharsets.UTF_8))
      .map(b => String.format("%02x", b.asInstanceOf[Object])).mkString
  }
}

File: src/main/kotlin/otkserver/SecurityUtils.kt
package otkserver

class MD5Utils {
	companion object {
		private val messageDigest = java.security.MessageDigest.getInstance("MD5")
    private val SALT = PropriedadeConfiguracoes("senha.salt")
    fun str2md5(senha: String) =
		  java.math.BigInteger(1,
			  messageDigest.digest("${senha}${SALT}".toByteArray(
			    java.nio.charset.StandardCharsets.UTF_8))
		  ).toString(16).padStart(32, '0')
	}
}
